lib/CMakeLists.txt

```diff
@@ -79,6 +80,7 @@ if(ENABLE_HIP)
   target_include_directories(proteus SYSTEM PRIVATE ${hip_INCLUDE_DIRS})
   target_compile_options(proteus PRIVATE -x hip)
   list(APPEND libs hip::amdhip64)
+  target_link_libraries(proteus PUBLIC hip::host)
```

Comment:
I believe this is required

Reply:
We do `list(APPEND libs hip::amdhip64)` which ends up in `target_link_libraries(proteus PRIVATE ${libs} "$<$<PLATFORM_ID:Darwin>:-undefined dynamic_lookup>") few lines later.
`

Reply:
PUBLIC since required by downstream users. This is an alternative to adding the `hip::host` dependency directly to the downstream user (which is what I initially had), since stuff gets sucked in through the `JitVariable` interface.

Since your restructuring and adding `__jit_push_variable` it may no longer be required.